lib/netsnmp/encryption/aes.rb

```diff
@@ -28,7 +28,7 @@ def encrypt(decrypted_data, engine_boots:, engine_time:)
       end
 
       def decrypt(encrypted_data, salt:, engine_boots:, engine_time:)
-        raise Error, "invalid priv salt received" unless (salt.length % 8).zero?
+        raise Error, "invalid priv salt received" unless !salt.empty && (salt.length % 8).zero?
```

Comment:
I think, it should be ".empty?".

Reply:
Would it possible to print the received salt. It would be helpful for debug.

Reply:
> I think, it should be ".empty?".

🤦  sorry about this, started the PR from github, didn't even realise.

> Would it possible to print the received salt. It would be helpful for debug.

The debug layer is a different problem altogether. I'd like to introduce debugging logs, to also help other reported bugs. Still struggling to find time to implement it though.